ocaml/typing/typeopt.ml

```diff
@@ -543,29 +543,41 @@ let value_kind env loc ty =
   with
   | Missing_cmi_fallback -> raise (Error (loc, Non_value_layout (ty, None)))
 
-let layout env loc sort ty =
-  match Layouts.Sort.get_default_value sort with
-  | Value -> Lambda.Pvalue (value_kind env loc ty)
+let[@inline always] layout_of_const_sort_generic ~value ~error
+  : Layouts.Sort.const -> _ = function
```

Comment:
I think `~value_kind` would be a better name than `~value` for the labeled parameter.

I am also tempted to rework this to take a parameter `~detailed_value_kind:(Location.t option)` (or something) and compute the value kind internally rather than passing in a `lazy` one.  But I think either way is fine if you prefer this.  It's very sad it can't be just a bool, in which case I would strongly prefer the other way.